--- origsrc/assogiate-0.2.1/libassogiate/misc/ucompose.cc	2007-06-12 13:55:00.000000000 -0500
+++ src/assogiate-0.2.1/libassogiate/misc/ucompose.cc	2007-07-15 21:45:58.437500000 -0500
@@ -161,14 +161,14 @@
 			output.insert(pos, rep);
 		}
 
-		os.str(std::wstring());
+		os.str(ustring());
 		/*os.clear();*/
 		++arg_no;
 	}
 }
 
 ustring
-UComposition::do_stringify(const std::wstring& src)
+UComposition::do_stringify(const ustring& src)
 {
 	return Glib::convert(std::string(reinterpret_cast<const char *>(src.data()),
 		src.size() * sizeof(wchar_t)), "UTF-8", "WCHAR_T");
--- origsrc/assogiate-0.2.1/libassogiate/misc/ucompose.hh	2007-06-12 00:21:48.000000000 -0500
+++ src/assogiate-0.2.1/libassogiate/misc/ucompose.hh	2007-07-15 21:45:58.437500000 -0500
@@ -85,16 +85,16 @@
 
 	void do_arg(const ustring &rep);
 
-	ustring do_stringify(const std::wstring& src);
+	ustring do_stringify(const ustring& src);
 
 	template<typename T>
 	std::string stringify(T obj)
 	{
 		os << obj;
-		return do_stringify(os.str());
+		return do_stringify(Glib::locale_to_utf8(os.str()));
 	}
 
-	std::wostringstream os;
+	std::ostringstream os;
 	int arg_no;
 
 	/**
--- origsrc/assogiate-0.2.1/src/Makefile.am	2007-06-11 23:19:56.000000000 -0500
+++ src/assogiate-0.2.1/src/Makefile.am	2007-07-15 21:42:04.578125000 -0500
@@ -10,6 +10,5 @@
 	type-dialog.hh type-dialog.cc \
 	main.hh main.cc
 
-assogiate_LDADD = $(ASSOGIATE_LIBS) \
-	../libassogiate/libassogiate.la
+assogiate_LDADD = ../libassogiate/libassogiate.la $(ASSOGIATE_LIBS)
 
