diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/assogiate-0.2.1/libassogiate/misc/ucompose.cc src/assogiate-0.2.1/libassogiate/misc/ucompose.cc
--- origsrc/assogiate-0.2.1/libassogiate/misc/ucompose.cc	2007-06-12 13:55:00.000000000 -0500
+++ src/assogiate-0.2.1/libassogiate/misc/ucompose.cc	2007-07-15 21:45:58.437500000 -0500
@@ -161,14 +161,14 @@
 			output.insert(pos, rep);
 		}
 
-		os.str(std::wstring());
+		os.str(ustring());
 		/*os.clear();*/
 		++arg_no;
 	}
 }
 
 ustring
-UComposition::do_stringify(const std::wstring& src)
+UComposition::do_stringify(const ustring& src)
 {
 	return Glib::convert(std::string(reinterpret_cast<const char *>(src.data()),
 		src.size() * sizeof(wchar_t)), "UTF-8", "WCHAR_T");
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/assogiate-0.2.1/libassogiate/misc/ucompose.hh src/assogiate-0.2.1/libassogiate/misc/ucompose.hh
--- origsrc/assogiate-0.2.1/libassogiate/misc/ucompose.hh	2007-06-12 00:21:48.000000000 -0500
+++ src/assogiate-0.2.1/libassogiate/misc/ucompose.hh	2007-07-15 21:45:58.437500000 -0500
@@ -85,16 +85,16 @@
 
 	void do_arg(const ustring &rep);
 
-	ustring do_stringify(const std::wstring& src);
+	ustring do_stringify(const ustring& src);
 
 	template<typename T>
 	std::string stringify(T obj)
 	{
 		os << obj;
-		return do_stringify(os.str());
+		return do_stringify(Glib::locale_to_utf8(os.str()));
 	}
 
-	std::wostringstream os;
+	std::ostringstream os;
 	int arg_no;
 
 	/**
diff -urN -x CYGWIN-PATCHES -x 'aclocal.m4*' -x autom4te.cache -x config.cache -x config.log -x config.status -x config.h -x config.h.in -x ABOUT-NLS -x Makefile.in.in -x Makevars.template -x '*SlackBuild*' -x '*.egg-info' -x '*.class' -x '*.pyc' -x '*.mo' -x '*.gmo' -x '*.orig' -x '*.rej' -x '*.spec' -x '*.temp' -x '*~' -x '*.stackdump' -x COPYING -x INSTALL -x compile -x config-ml.in -x config.guess -x config.sub -x depcomp -x elisp-comp -x install-sh -x ltmain.sh -x mdate-sh -x missing -x mkinstalldirs -x py-compile -x symlink-tree -x texinfo.tex -x ylwrap -x Makefile.in -x makefile.in -x configure -x omf.make -x xmldocs.make -x gnome-doc-utils.make -x gnome-doc-utils.m4 -x intltool.m4 -x intltool-extract -x intltool-extract.in -x intltool-merge -x intltool-merge.in -x intltool-update -x intltool-update.in origsrc/assogiate-0.2.1/src/Makefile.am src/assogiate-0.2.1/src/Makefile.am
--- origsrc/assogiate-0.2.1/src/Makefile.am	2007-06-11 23:19:56.000000000 -0500
+++ src/assogiate-0.2.1/src/Makefile.am	2007-07-15 21:42:04.578125000 -0500
@@ -10,6 +10,5 @@
 	type-dialog.hh type-dialog.cc \
 	main.hh main.cc
 
-assogiate_LDADD = $(ASSOGIATE_LIBS) \
-	../libassogiate/libassogiate.la
+assogiate_LDADD = ../libassogiate/libassogiate.la $(ASSOGIATE_LIBS)
 
